### PR TITLE
[improve][workflow] Add a job in document bot to auto generate config docs

### LINKED ISSUE #13453: [Website] generate HTML for Pulsar configuration pages
Generate HTML for the components in the red box:
image

Details see [here](https://docs.google.com/spreadsheets/d/1P6tePGmU1lEhGKY_hE3JWgJ1rUTBCmW7AshDICJZr3o/edit#gid=1030826289)

### PR DESCRIPTION
**Pull request by SignorMercurio**
Fix #13453

### Motivation

In #16627, we can now generate config docs by manually running a script after changing specific code files that contains annotations. This PR adds a job in the existing document bot to automate this step.

### Modifications

~A new job `doc-gen` is added in `ci-documentbot.yml`~ A new workflow `doc-auto-gen` is added which will only trigger when there're new pushes to `master` updating the 4 code files that contains configuration annotations.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)

CI adjustment.

- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@maxsxu @lhotari 
could you please review this PR from the technical perspective? Thank you!

**Review by urfreespace — CHANGES_REQUESTED** [review]
@SignorMercurio  `ci-documentbot` is not associated with `auto-generate docs ci` pls make a new independent ci workflow for it /cc @Anonymitaet

**Review by maxsxu — COMMENTED** [review]
LGTM. Adding a test link would be better.

**Review by tisonkun — CHANGES_REQUESTED** [review]
-1

Apache generally bans non-committers/third-party actors to have changes merged without provenance.

We can let the bot create a pull request, or add a script to help developers generate config docs. But the actual task should be checked before merging into master or any release branch.

**Comment by tisonkun** [comment]
You may think of code style guards: if someone sends a patch violent checkstyle rules, we fail the CI and ask the contributor fix it. We don't let a bot to automatically push a commit there.

While checkstyle doesn't provide autofix mechanism, switch to spotless can provide one. But every Apache project let the contributor to apply that automation instead of by a bot.

You can replace code style with up-to-date config docs to understand the comment.

**Comment by SignorMercurio** [comment]
@tisonkun Understood. Actually a simple script to generate these docs are already added. Maybe we shouldn't add a workflow at all (since we need to check for the updates manually)?

cc @Anonymitaet

File: .github/workflows/doc-auto-gen.yml
Comment (dave2wave) [reply .github/workflows/doc-auto-gen.yml:74]: @tisonkun is correct. If these changes are on the master or other protected branch then these are likely disallowed.

**Comment by Anonymitaet** [comment]
@tisonkun thanks for your reminder! 

We've discussed it and we're researching another way to implement it. 

For example,  make [Pulsar configuration pages](https://pulsar.apache.org/docs/next/reference-configuration-broker) as independent pages (like https://pulsar.apache.org/tools/pulsar-admin/2.11.0-SNAPSHOT/) rather than `.md` files. In this way, we do not need to create such a workflow.

**Pull request closed without merging**
